pyfr/integrators/base.py

```diff
@@ -58,6 +58,29 @@ def __init__(self, backend, rallocs, mesh, initsoln, cfg):
         # Abort computation
         self.abort = False
 
+        # Smoothly step to target time in the last near_t steps
+        self.fact = self.cfg.getfloat('solver-time-integrator', 'dt-fact', 0.9)
+        self.fact_max = 1.001
+        self.dt_in = cfg.getfloat('solver-time-integrator', 'dt')
+        self.dt_near = None
+
+    def adjust_dt(self, t):
```

Comment:
If I'm not mistaken, can this function not be replaced with
```python
def adjust_dt(self, t):
    if (tdiff := t - self.tcurr) < self.dt*self*fact_max:
        self.dt = tdiff
```
or is there some edge case I've missed?

Reply:
The logic you suggested does not consider a gradual change of `self.dt`. When we are a few time steps away from the target `dt`, the third `if` condition sets a `dt` value we would take as we approach to the target `dt`. Also, I have used a `self.dt_near` variable to decrease the number of times we change `dt`, and set it to `None` when we are done with using the `dt_near` for our target `t`.


Test it here:

https://colab.research.google.com/drive/1upThO_hbxr25ThikrYOAJa-IHNGjq-pD?usp=sharing